aws-janitor/resources/set_test.go

```diff
@@ -125,11 +125,14 @@ func TestMarkCreationTimes(t *testing.T) {
 				ExpectedFirstSeen: threeHoursAgo,
 			},
 			{
-				// Some resources only have times that can potentially reset to newer times (e.g. startup date)
+				// Some resources might be created again with the same name, and we don't want to delete those.
+				// Note that some resources don't record a creation timestamp,
+				// and only have times that can potentially reset to newer times (e.g. startup date),
+				// potentially we will leak those, but if the data keeps increasing that is taken as a usage signal.
```

Comment:
I don't follow this last part. Why are we ok with leaking these boskos resources?

Reply:
So I'd argue that we shouldn't pass creationTimestamp unless it is a creation timestamp, but that felt like a bigger change.

Right now, there are a few places where pass a different value instead of nil or the creationTimestamp:

* Instances pass the LaunchTime
* Disks pass the AttachTime
* Disk Snapshots pass the StartTime
* SQS passes what looks like a creation timestamp, but parsed from an attribute


These values are all within a few seconds of the actual creation time (I assume), except for disks or instances where I think we can reattach or relaunch them and "restart the clock".  Then these resources will not be cleaned up, but only if they are continually being reattached or relaunched .... in which case they are in use, and thus I think we don't _want_ to clean them up.

Reply:
Ok that makes sense, but are such resources leaked permanently or will they later change to a state where we will clean them up? This sounds good as long as the resources will eventually be recovered and aren't permanently leaked.
Sorry, I'm not familiar with this code.

Reply:
They'll be deleted after the TTL (4 hours, IIRC).  They only don't get cleaned up if we have jobs that keep deleting and recreating them, but that's not really leaking.